### ISSUE TITLE
Deprecation warnings while installing Zend framework skeleton app 3.1.0 

### ISSUE THREAD

**Issue by prbt2016** [zendframework/ZendSkeletonApplication#403]
Hello ,
I was trying to install Zend framework skeleton application (version 3.1.0) , downloaded from here 
https://github.com/zendframework/ZendSkeletonApplication/releases/. However when i try to install it via composer i.e( php composer.phar install)  i get the following deprecation warnings : 

  - Installing zendframework/zend-component-installer (0.3.0) Loading from cache
Deprecation Notice: The callback Zend\ComponentInstaller\ComponentInstaller::onPostPackageInstall declared at {PATH}/{TO}vendor/zendframework/zend-component-installer/src/ComponentInstaller.php accepts a Composer\Script\PackageEvent but post-package-install events use a Composer\Installer\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar://{PATH}/{TO}composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:317
  - Installing zendframework/zend-stdlib (3.0.1) Loading from cache
Deprecation Notice: The callback Zend\ComponentInstaller\ComponentInstaller::onPostPackageInstall declared at {PATH}/{TO}vendor/zendframework/zend-component-installer/src/ComponentInstaller.php accepts a Composer\Script\PackageEvent but post-package-install events use a Composer\Installer\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar://{PATH}/{TO}composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:317.

What could be the issue?.For your kind information i am using php 5.6.Are these warnings critical ?

**Comment by prbt2016** [comment]
Awaiting your kind reply.

**Comment by Ocramius** [comment]
Weird that you got `0.3.0` - current stable version is `0.6.*` as per https://github.com/zendframework/zend-component-installer

**Comment by webimpress** [comment]
@Ocramius if he used `composer install`, it means all packages are installed in versions locked in `composer.lock` file, and - for `zend-component-installer` there is 0.3.0 version:
https://github.com/zendframework/ZendSkeletonApplication/blob/master/composer.lock#L38-L39

@prbt2016 please try to run `composer update`, once `zend-component-installer` is updated then all should be fine.

**Comment by alextech** [comment]
Should we update lock file in this repository? Just ran into this with `create-project -s dev zendframework/skeleton-application`.

**Comment by prbt2016** [comment]
Hello,
**@ webimpress**  : As per your suggestion I tried composer update , still it gives the following warnings : 
   - Installing zendframework/zend-component-installer (0.3.1) Loading from cache
Deprecation Notice: The callback Zend\ComponentInstaller\ComponentInstaller::onPostPackageInstall declared at {{PATH}}/{{TO}}/vendor/zendframework/zend-component-installer/src/ComponentInstaller.php accepts a Composer\Script\PackageEvent but post-package-install events use a Composer\Installer\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar://{{PATH}}/{{TO}}/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:317
  - Installing zendframework/zend-skeleton-installer (0.1.3) Loading from cache
Deprecation Notice: The callback Zend\ComponentInstaller\ComponentInstaller::onPostPackageInstall declared at {{PATH}}/{{TO}}/vendor/zendframework/zend-component-installer/src/ComponentInstaller.php accepts a Composer\Script\PackageEvent but post-package-install events use a Composer\Installer\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar://{{PATH}}/{{TO}}/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:317.

Is there any other way to fix this ?.awaiting your kind response.

**Comment by webimpress** [comment]
@prbt2016 This is odd... Not sure why `zend-component-installer` is not updated then to version [0.6.0](https://github.com/zendframework/zend-component-installer/releases/tag/0.6.0). Can you try update your `composer.json` to point this version and run again `composer update`, please?

@alextech 
> Should we update lock file in this repository? Just ran into this with `create-project -s dev zendframework/skeleton-application`.

Yeah, I think it should be updated in the repository, and `zend-component-installer 0.6.0` should be used. Thanks!

**Comment by prbt2016** [comment]
@webimpress 
Do you mean to say ?.I should edit line no 16 in composer.json that reads 
"zendframework/zend-component-installer": "^1.0 || ^0.3 || ^1.0.0-dev@dev".?.How should i edit this to make it point to 0.6.0 ?.Awaiting your response.

**Comment by webimpress** [comment]
@prbt2016 try this please:
`"zendframework/zend-component-installer": "^1.0 || ^0.6 || ^1.0.0-dev@dev"`
and run `composer update`.

**Comment by prbt2016** [comment]
@webimpress  : 
Did as you suggested.But However after i run php composer.phar update .It throws the following error : 
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - zendframework/zend-skeleton-installer 0.1.3 requires zendframework/zend-component-installer ^1.0 || ^0.3 || ^1.0.0-dev@dev -> satisfiable by zendframework/zend-component-installer[1.0.x-dev].
    - zendframework/zend-skeleton-installer 1.0.x-dev requires zendframework/zend-component-installer ^1.0 || ^0.3 || ^1.0.0-dev@dev -> satisfiable by zendframework/zend-component-installer[1.0.x-dev].
    - Conclusion: don't install zendframework/zend-component-installer 1.0.x-dev
    - Installation request for zendframework/zend-skeleton-installer ^1.0 || ^0.1.3 || ^1.0.0-dev@dev -> satisfiable by zendframework/zend-skeleton-installer[0.1.3, 1.0.x-dev].

**Comment by webimpress** [comment]
@prbt2016 Thanks, we need then also update `zendframework/zend-skeleton-installer`. Currently there is no PR to support `zend-component-installer 0.6`. I didn't notice it before. I'll create PR later on and hopefully ZendSkeletonApplication will be updated soon.

**Comment by prbt2016** [comment]
@webimpress : 
And for your kind reference these are those warnings that i get while executing 
"php composer.phar install".

I have uploaded a file for your reference with those whole warnings.
[warnings.txt](https://github.com/zendframework/ZendSkeletonApplication/files/738658/warnings.txt).Please take a look.

**Comment by webimpress** [comment]
@prbt2016 I've tried to install skeleton application using:
`composer create-project -sdev zendframework/skeleton-application path/to/install`

And I have got the same deprecation notices as you:

```
Installing zendframework/skeleton-application (dev-master 941da45b407e4f09e264f000fb537928badb96ed)
  - Installing zendframework/skeleton-application (dev-master master) Cloning master from cache
Created project in zend-skap
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 17 installs, 0 updates, 0 removals
  - Installing zendframework/zend-component-installer (0.3.0) Downloading: 100%
Deprecation Notice: The callback Zend\ComponentInstaller\ComponentInstaller::onPostPackageInstall declared at path/to/install/vendor/zendframework/zend-component-installer/src/ComponentInstaller.php accepts a Composer\Script\PackageEvent but post-package-install events use a Composer\Installer\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar://usr/local/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:320
[...]

Generating autoload files

    Do you want a minimal install (no optional packages)? Y/n
Y
    Removing optional packages from composer.json
    Updating composer.json
  - Removing zendframework/zend-skeleton-installer (0.1.3)
Removing zendframework/zend-skeleton-installer...
    Removed plugin zendframework/zend-skeleton-installer.
    Removing from composer.json
    Complete!
Do you want to remove the existing VCS (.git, .svn..) history? [Y,n]?
> zf-development-mode enable
You are now in development mode.
```

After that I was able to do:
```
cd path/to/install
composer update

Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 10 updates, 0 removals
  - Updating zendframework/zend-component-installer (0.3.0 => 0.6.0) Downloading: 100%
  - Updating zendframework/zend-stdlib (3.0.1 => 3.1.0) Downloading: 100%
  - Updating zendframework/zend-eventmanager (3.0.1 => 3.1.0) Downloading: 100%
  - Updating zendframework/zend-view (2.8.0 => 2.8.1) Downloading: 100%
  - Updating zendframework/zend-servicemanager (3.1.0 => 3.2.0) Downloading: 100%
  - Updating zendframework/zend-validator (2.8.1 => 2.8.2) Downloading: 100%
  - Updating zendframework/zend-escaper (2.5.1 => 2.5.2) Downloading: 100%
  - Updating zendframework/zend-http (2.5.4 => 2.5.5) Downloading: 100%
  - Updating zendframework/zend-mvc (3.0.1 => 3.0.4) Downloading: 100%
  - Updating zfcampus/zf-development-mode (3.0.0 => 3.1.0) Downloading: 100%
Writing lock file
Generating autoload files
```
without any deprecation notices.

Ofc will be nice to eliminate these deprecation notice on create-project, but finally all is working fine.
Please notice above that package `zendframework/zend-skeleton-installer` is removed after successful installation and on `composer update` package `zendframework/zend-component-installer` is updated to version `0.6.0`.

My current `composer.json` is as below:

```json
{
    "name": "zendframework/skeleton-application",
    "description": "Skeleton Application for Zend Framework zend-mvc applications",
    "type": "project",
    "license": "BSD-3-Clause",
    "keywords": [
        "framework",
        "mvc",
        "zf"
    ],
    "homepage": "http://framework.zend.com/",
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "php": "^5.6 || ^7.0",
        "zendframework/zend-component-installer": "^1.0 || ^0.6 || ^1.0.0-dev@dev",
        "zendframework/zend-mvc": "^3.0.1",
        "zfcampus/zf-development-mode": "^3.0"
    },
    "autoload": {
        "psr-4": {
            "Application\\": "module/Application/src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "ApplicationTest\\": "module/Application/test/"
        }
    },
    "extra": [],
    "scripts": {
        "cs-check": "phpcs",
        "cs-fix": "phpcbf",
        "development-disable": "zf-development-mode disable",
        "development-enable": "zf-development-mode enable",
        "development-status": "zf-development-mode status",
        "post-create-project-cmd": [
            "@development-enable"
        ],
        "serve": "php -S 0.0.0.0:8080 -t public/ public/index.php",
        "test": "phpunit"
    }
}
```

I've tried the same with cloning the main repository and `composer install`. I have got the same results, so deprecation notices on install and removed `zend-skeleton-installer` at the end, then `composer update` updates `zend-component-installer` (to version `0.6.0`) and no any more deprecation notices.

Please try again and let me know if you still have any issues.

Again, deprecation notices should be eliminated with next version of ZendSkeletonApplication with  `zend-composer-installer 0.6.0`, but it's not a big issue as long as all is working as expected.

**Comment by prbt2016** [comment]
@webimpress  : 
Thank you for your kind response.Yes even i got the same result as you after i tried php composer.phar install.Okay i got the deprecation warnings .And then i manually changed the _zend-component-installer_ version to 0.6.0 .Then i executed php composer.phar update.Here's the result of what i got  : 

Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 10 updates, 0 removals
  - Updating zendframework/zend-component-installer (0.3.0 => 0.6.0) Loading from cache
  - Updating zendframework/zend-stdlib (3.0.1 => 3.1.0) Loading from cache
  - Updating zendframework/zend-eventmanager (3.0.1 => 3.1.0) Loading from cache
  - Updating zendframework/zend-view (2.8.0 => 2.8.1) Loading from cache
  - Updating zendframework/zend-servicemanager (3.1.0 => 3.2.0) Loading from cache
  - Updating zendframework/zend-validator (2.8.1 => 2.8.2) Loading from cache
  - Updating zendframework/zend-escaper (2.5.1 => 2.5.2) Loading from cache
  - Updating zendframework/zend-http (2.5.4 => 2.5.5) Loading from cache
  - Updating zendframework/zend-mvc (3.0.1 => 3.0.4) Loading from cache
  - Updating zfcampus/zf-development-mode (3.0.0 => 3.1.0) Loading from cache
Writing lock file
Generating autoload files.Thank you so much .Before this i was trying to update the dependencies first and then run composer install.That's why it was failing.

It will be great if those notices are eliminated while composer install itself.Thanks for your great response.Would like to ask you that when shall the next version of Zend skeleton be released approximately?

**Comment by prbt2016** [comment]
@webimpress : 
It would be really great and appreciable  if you could update the newer version with those changes so that while composer install itself  , all those packages will be the updated ones.Because for me as a lay user it's not  feasible to do the  technical  tweaks .awaiting your kind reply .

**Comment by webimpress** [comment]
@prbt2016 I don't have push permission to this repository so I can't update it. I spoke with @weierophinney and he is going have a look on it as soon as he can.

**Comment by prbt2016** [comment]
@webimpress :That's  good.Thanks for your  good support .

**Comment by albrin** [comment]
I'm sorry, but the Zend Framework is "Enterprise Ready"?

Maybe I do not understand something in this dependency and this bug.

The issue opened 24 days ago. During this time, no one was pulling the request and merge?

It also does not require a lot of time?

In https://github.com/zendframework/ZendSkeletonApplication/blob/master/composer.json changed from:
```bash
"zendframework/zend-component-installer": "^1.0 || ^0.3 || ^1.0.0-dev@dev",
```
to:
```bash
"zendframework/zend-component-installer": "^1.0 || ^0.6 || ^1.0.0-dev@dev",
```
And in https://github.com/zendframework/zend-skeleton-installer/blob/master/composer.json changed from:
```bash
"zendframework/zend-component-installer": "^1.0 || ^0.3 || ^1.0.0-dev@dev"
```
to:
```bash
"zendframework/zend-component-installer": "^1.0 || ^0.6 || ^1.0.0-dev@dev"
```

I love this Framework! But the update for some reason do not often go. I would start to contribute, but I do have a little experience with the github.

I have big plans for this framework in their projects in the future (sites, services, SPA, CMS and other)! 

You are doing a great job! Thank you so much!
But five years of work with this framework I have the impression in low activity.

P.S.: I would also like to work on even better performance this framework was made. And then he will not be equal. I also could not find the plans for further development of the project.

**Comment by prbt2016** [comment]
@webimpress : Any updates regarding the next version to be released and the above  issue being fixed ?
.Waiting for your kind reply.

**Comment by Xerkus** [comment]
Fixed in #412
